src/main/java/com/sda10/carrental/service/CarService.java

```diff
@@ -34,9 +38,26 @@ public Car updateCar(Long id, Car car){
         }
     }
 
-    public void deleteCar(Long id){
-        Car existingCar=carRepository.findById(id).get();
+    public void deleteCar(Long id) {
+        Car existingCar = carRepository.findById(id).get();
 
         carRepository.delete(existingCar);
     }
+
+    public List<Car> carFilters() {
+
+        List<Car> filterList = new ArrayList<>();
+
+        return filterList.stream()
+                .filter(car -> car.getMake().equals(CarDto.carDto().make))
+                .filter(car -> car.getModel().equals(CarDto.carDto().model))
+                .filter(car -> car.getBodyType().equals(CarDto.carDto().bodyType))
+                .filter(car -> car.getYearOfProduction().equals(CarDto.carDto().yearOfProduction))
+                .filter(car -> car.getColor().equals(CarDto.carDto().color))
+                .filter(car -> car.getMileage().equals(CarDto.carDto().mileage))
+                .filter(car -> car.getStatus().equals(CarDto.carDto().status))
+                .filter(car -> car.getAmount().equals(CarDto.carDto().amount))
```

Comment:
Aproape corect: 
in loc de CarDto.carDto() care iti creeaza o instanta noua de carDto de fiecare data cand il apelezi trebuie sa folosesti un parametru care va fi primit de metoda carFilters si acolo primesti filtrele. De asemenea trebuie sa te gandesti ce faci in cazul in care nu este specificata valoarea pentru unul din filtre, daca vei compara null cu o valoare existenta iti va da fals si elementul nu va fi returnat.